hel_django_auditlog_extra/README.md

```diff
@@ -0,0 +1,330 @@
+    - Raise an error if any models are not configured when
+    `AUDITLOG_INCLUDE_ALL_MODELS` is enabled.
+
+    This helps ensure that all models are either explicitly included or excluded
+    from audit logging, preventing accidental omissions.
+
+NOTE: The `AuditLogConfigurationHelper` can be used only after all the apps are ready.
+
+Example usage: Use when the audit log registry is already configured...
+
+```python
+AuditLogConfigurationHelper.raise_error_if_unconfigured_models()
+```
+
+##### Initialization examples for `AuditLogConfigurationHelper`:
+
+- One place to call this helper function would be in the ready-function of the last app's configuration, when all the models are already registered. You can even add an `apps.py` file into your main Django project folder and then add the project app to the list of the `INSTALLED_APPS` in project's `settings.py`. The important thing is that it should be the last one with models.
+
+  **`apps.py`**:
+
+  ```python
+  from django.apps import AppConfig
+
+  class MainProjectConfig(AppConfig):
+      name = "mainproject"
+
+      def ready(self):
+          from hel_django_auditlog_extra.utils import AuditLogConfigurationHelper
+
+          AuditLogConfigurationHelper.raise_error_if_unconfigured_models()
+  ```
+
+  **`settings.py`**:
+
+  ```python
+  INSTALLED_APPS = [
+      "django.contrib.auth",
+      "django.contrib.contenttypes",
+      "django.contrib.sessions",
+      "django.contrib.messages",
+      "django.contrib.staticfiles",
+      "auditlog",
+      # local apps
+      "events",
+      "hel_django_auditlog_extra.apps.HelDjangoAuditLogExtraConfig",
+      "mainproject",
+      "django_cleanup.apps.CleanupConfig",  # This must be included last
+  ]
+  ```
+
+- Another way to achieve this would be using a `post_migrate` signal receiver, which is called after the migration process (`python manage.py migrate`) is done:
+
+  **`signals.py`**:
+
+  ```python
+  from django.apps import apps
+  from django.db.models.signals import post_migrate
+  from django.dispatch import receiver
+
+  from hel_django_auditlog_extra.utils import AuditLogConfigurationHelper
+
+
+  @receiver(post_migrate)
+  def check_audit_log_configuration(sender, **kwargs):
+      """
+      Signal receiver to check audit log configuration after all apps are migrated.
+      """
+      if apps.ready:
+          AuditLogConfigurationHelper.raise_error_if_unconfigured_models()
+  ```
+
+  Remember that signals and receivers needs to be connected (for example in `apps.py`):
+
+  ```python
+  from django.apps import AppConfig
+  from django.core.signals import request_finished
+
+
+  class MyAppConfig(AppConfig):
+      ...
+
+      def ready(self):
+          # Implicitly connect signal handlers decorated with @receiver.
+          from . import signals
+
+          # Explicitly connect a signal handler.
+          request_finished.connect(signals.my_callback)
+  ```
```

Comment:
Is this needed? Looking at the https://docs.djangoproject.com/en/5.1/topics/signals/#listening-to-signals it does mention this as the "manual connect route" but then using the `receiver` decorator as an alternative for this. And the `receiver` decorator is documented already above.

If you feel this is needed then it could be made clear that this is an alternative to using the `receiver` decorator for connecting signals to receivers.

Reply:
`from . import signals` is still needed, so the receiver decorator is read. If the decorator is not used, then also the `connect` needs to be called for the signal. Or did you mean something else?

Reply:
I was just reading the README and thought the train-of-thought was a bit off, that's what I meant by my comment. So, basically IMHO how this whole section with how the signals can be connected (using the decorator and not) reads as a whole could use some tuning. As a whole especially, that's my point, not just as separate paragraphs.